src/library_webgl2.js

```diff
@@ -730,7 +730,7 @@ var LibraryWebGL2 = {
       return;
     }
     var ret = GLctx.getSyncParameter(GL.syncs[sync], pname);
-    {{{ makeSetValue('length', '0', 'ret', 'i32') }}};
+    {{{ makeSetValue('values', '0', 'ret', 'i32') }}};
     if (ret !== null && length) {{{ makeSetValue('length', '0', '1', 'i32') }}}; // Report a single value outputted.
```

Comment:
Good catch! Indeed a copy-paste typo or something here.

Reading the spec: https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glGetSynciv.xhtml
```
On success, glGetSynciv replaces up to bufSize integers in values with the corresponding property values of the object being queried. The actual number of integers replaced is returned in the variable whose address is specified in length. If length is NULL, no length is returned.

If an error occurs, nothing will be written to values or length.
```

So the correct code here should be:

```js
var ret = GLctx.getSyncParameter(GL.syncs[sync], pname);
var notNull = ret !== null;
if (notNull) {{{ makeSetValue('values', '0', 'ret', 'i32') }}};
if (length) {{{ makeSetValue('length', '0', 'notNull', 'i32') }}}; // Report the number of values outputted. (coerce bool to 0/1 value)
```
What do you think?

Reply:
Yes, good point! my code hadn't handled that error case. Per the spec it looks like neither values nor length should be written to if an error occurs, so looks like both writes should be under that notNull if branch. Just amended my commit again to reflect these changes.